### PR TITLE
rustPackages.rls: Fix 1.55 build on Darwin

### LINKED ISSUE #140229: rls-1.55 on nixpkgs-unstable fails to build on macOS
Building `nixpkgs.rustPackages.rls` on Darwin, building derivative `/nix/store/pcx9pb2jx3i7ipi2w9vdsl68ngvxnsrn-rls-1.55.0.drv`, compilation fails with:
```
  cargo:warning=curl/lib/hostip.c:73:10: fatal error: 'SystemConfiguration/SCDynamicStoreCopySpecific.h' file not found
  cargo:warning=#include <SystemConfiguration/SCDynamicStoreCopySpecific.h>
  cargo:warning=         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=1 error generated.
  exit status: 1
```

### PR DESCRIPTION
**Pull request by jwiegley**
###### Motivation for this change

fixes #140229 

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
@ofborg eval

**Comment by Mic92** [comment]
@ofborg eval

**Comment by figsoda** [comment]
can you squash the commits into `rls: fix darwin build`?

File: pkgs/development/compilers/rust/rls/default.nix [hunk 1]
```diff
@@ -30,7 +30,7 @@ rustPlatform.buildRustPackage rec {
 
   nativeBuildInputs = [ pkg-config cmake makeWrapper ];
   buildInputs = [ openssh openssl curl zlib libiconv rustPlatform.rust.rustc.llvm ]
-    ++ (lib.optionals stdenv.isDarwin [ CoreFoundation Security ]);
+    ++ (lib.optionals stdenv.isDarwin [ CoreFoundation Security SystemConfiguration ]);
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/rust/rls/default.nix:33]:
```suggestion
    ++ lib.optionals stdenv.isDarwin [ CoreFoundation Security SystemConfiguration ];
```

**Pull request merged**